packages/create-cloudflare/src/frameworks/angular/templates/server.ts

```diff
@@ -0,0 +1,34 @@
+import { renderApplication } from '@angular/platform-server';
+import bootstrap from './src/main.server';
+
+interface Env {
+  ASSETS: { fetch: typeof fetch };
+}
+
+// We attach the Cloudflare `fetch()` handler to the global scope
+// so that we can export it when we process the Angular output.
+// See tools/bundle.mjs
+async function workerFetchHandler(request: Request, env: Env) {
+  const url = new URL(request.url);
+  console.log('render SSR', url.href);
+
+  // Get the root `index.html` content.
+  const indexUrl = new URL('/', url);
```

Comment:
Is there a way to use an index file which is not part of assets? IE: not accessible from the browser. For context, this is because for SSR we have a dedicated page in `server/index.server.html` which does not have critical css inlined.

Reply:
We could inline the content of the file as an import here. Something like:

```ts
import document from "../server/index.html"

...


const content = await renderApplication(bootstrap, {
  document, 
  url: url.pathname
})
```

Reply:
I tried this and got it working for deployed apps and local app, but unfortunately we get a conflict in the allowed path name which means I couldn't get it working for both.

Since the `index.server.html` is generated by Angular and only available post-build we have to convince TS and esbuild that this import is external. We can do this by using `@ts-ignore` for TS and wrapping the import in a try-catch for esbuild.

```ts
let indexHTM: any;
try {
  // @ts-ignore
  indexHTML = await import("index.server.html");
} catch {}
const document = index.default
```

The deployed version requires the import to be `"index.server.html"` but for `wrangler pages dev` it must be `"./index.server.html"`.

I couldn't find a good way to avoid this conflict.